File: Services/BooksManager/BM.Services.BooksManager.Core/Models/Entities/Book.cs
using BM.Services.BooksManager.Core.Models.API;
using BM.Services.BooksManager.Core.Models.Relationships;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BM.Services.BooksManager.Core.Models.Entities
{
    public class Book
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string CoverImagePath { get; set; }
        public ICollection<BookAuthor> Authors { get; set; }

        public static implicit operator BookApiModel(Book model)
        {
            if (model == null) return null;
            return new BookApiModel
            {
                Id = model.Id,
                Title = model.Title,
                Description = model.Description,
                CoverImage = model.CoverImagePath,
                Authors = model.Authors,
            };
        }
    }
}

File: Frontend/booksmanager-app/src/app/models/book.ts

import { BookAuthor } from "./book-author";

export class Book {
    id!: number;
    title!: string;
    description!: string;
    authors!: BookAuthor[];
    coverImage!: string;
  }

  export class CreateBookCommand{
    title!: string;
    description!:string;
    coverImagePath!:string;
    file!:any;
    authorsIds:number[]=[];
  }

  export class UpdateBookCommand{
    id!:number;
    title!: string;
    description!:string;
    authorsIds:number[]=[];
  }